### PR TITLE
[FLINK-19281][Table SQL / API]LIKE cannot recognize full table path

### PR DESCRIPTION
**Pull request by zhuxiaoshang**
## What is the purpose of the change

to support full table path in flink sql ddl
for example, if we have a table whose full path is default_catalog.default_database.my_table1, and the following DDL will fail

```
create table my_table2 
like default_catalog.default_database.my_table1
```

## Brief change log
  - to support full table path in flink sql ddl

## Verifying this change

This change added tests 

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by zhuxiaoshang** [comment]
@flinkbot run azure

**Comment by zhuxiaoshang** [comment]
@flinkbot run azure

**Comment by zhuxiaoshang** [comment]
@libenchao Could you please have a review.

**Review by libenchao — COMMENTED** [review]
The changes LGTM generally, just one minor comment.

@wuchong Do you have time to have a look?

File: flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlCreateTableConverter.java [hunk 1]
```diff
@@ -141,7 +141,7 @@ private CatalogTable createCatalogTable(SqlCreateTable sqlCreateTable) {
 
 	private CatalogTable lookupLikeSourceTable(SqlTableLike sqlTableLike) {
 		UnresolvedIdentifier unresolvedIdentifier = UnresolvedIdentifier.of(sqlTableLike.getSourceTable()
-			.toString());
+			.names);
```
[the remark this answers is not in the document]
Comment (zhuxiaoshang) [reply flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlCreateTableConverter.java:144]: of course.

[the remark this answers is not in the document]
Comment (zhuxiaoshang) [reply flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlCreateTableConverter.java]: Done.

**Review by wuchong — APPROVED** [review]
LGTM.

Feel free to merge this @libenchao .

**Pull request closed without merging**